packages/libs/eda/src/lib/core/components/visualizations/implementations/NetworkVisualization.tsx

```diff
@@ -0,0 +1,561 @@
+    const scaleY = scaleLinear()
+      .domain([minYPosition, maxYPosition])
+      .range([100, plotContainerStyles.height - 100]);
+
+    // Create map that will adjust each link's weight to find a stroke width that spans an appropriate range for this viz.
+    const weightToStrokeWidthMap = scaleLinear()
+      .domain([minDataWeight, maxDataWeight])
+      .range([MIN_STROKE_WIDTH, MAX_STROKE_WIDTH]);
+
+    // Assign color to links.
+    // Color palettes live here in the frontend, but the backend decides how to color links (ex. by sign of correlation, or avg degree of parent nodes).
+    // So we'll make assigning colors generalizable by mapping the values of the links.color prop to the palette. As we add
+    // different ways to color links in the future, we can adapt our checks and error messaging.
+    const uniqueLinkColors = uniq(
+      data.value?.network.data.links.map(
+        (link) => link.color?.toString() ?? DEFAULT_LINK_COLOR_DATA
+      )
+    );
+    if (uniqueLinkColors.length > twoColorPalette.length) {
+      throw new Error(
+        `Found ${uniqueLinkColors.length} link colors but expected only ${twoColorPalette.length}.`
+      );
+    }
+    // The link color sent from the backend should be either '-1' or '1', but we'll allow any two unique values. Assigning the domain
+    // in the following way preserves "1" getting mapped to the second color in the palette, even if it's the only
+    // unique value in uniqueLinkColors.
+    const linkColorScaleDomain = uniqueLinkColors.every((val) =>
+      ['-1', '1'].includes(val)
+    )
+      ? ['-1', '1']
+      : uniqueLinkColors;
+    const linkColorScale = scaleOrdinal<string>()
+      .domain(linkColorScaleDomain)
+      .range(twoColorPalette); // the output palette may change if this visualization is reused in other contexts (ex. not a correlation app).
+
+    // Find display labels
+    const nodesWithLabels = data.value.network.data.nodes.map((node) => {
+      // node.id is the entityId.variableId
+      const displayLabel = fixVarIdLabel(
+        node.id.split('.')[1],
+        node.id.split('.')[0],
+        entities
+      );
+
+      return {
+        ...node,
+        x: scaleX(Number(node.x)),
+        y: scaleY(Number(node.y)),
+        id: node.id,
+        label: displayLabel,
+        // the following attempts to place the label in a "reasonable" position
+        // on the plot. So if a node is far on the left side, the label will be on the left side.
+        // This simple solution attempts to avoid overlapping labels and give us a cheap, clean-ish plot.
+        labelPosition:
+          scaleX(Number(node.x)) >
+          (plotContainerStyleOverrides?.width ?? 900) / 2
+            ? 'right'
+            : ('left' as LabelPosition),
+      };
+    });
+
+    // set initial visible node labels
+    const nodeLabels = nodesWithLabels.flatMap((node) => {
```

Comment:
For clarification, let's call this one `defaultNodeLabels` or `initialNodeLabels`. What do you think? I'm hoping to make it easier for someone searching for "nodeLabels" in the future :)

Reply:
@asizemore sounds good 👍 Will do that.